src/descriptor/mod.rs

```diff
@@ -259,6 +260,19 @@ pub enum DescriptorType {
     Tr,
 }
 
+impl DescriptorType {
```

Comment:
I would also add a `is_taproot()` method, even though it will be unused here in miniscript. I'm sure somebody will find that useful!

Reply:
Maybe `segwit_version(&self) -> Option<u8>` would be the most general?